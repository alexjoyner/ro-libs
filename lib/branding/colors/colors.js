'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
var COLORS = exports.COLORS = {
  'primaryLight': '#74b9ff',
  'primary': '#0984e3',
  'successLight': '#55efc4',
  'success': '#00b894',
  'warningLight': '#ffeaa7',
  'warning': '#fdcb6e',
  'dangerLight': '#ff7675',
  'danger': '#d63031'
};