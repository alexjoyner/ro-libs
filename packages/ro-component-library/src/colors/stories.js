import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import { Panel } from '../Panel';
import { colors } from '.';
import { CenteredContent } from '../CenteredContent';
import { getTextContrastColor } from '../utils/getTextContrastColor';

storiesOf('Depreciated', module)
  .add('Colors', () => (
    <div>
      <Panel>
        <CenteredContent>
          <h2>Component library color pallete</h2>
          {Object.keys(colors).map(key => (
            <div style={{
              height: '100px',
              width: '100%',
              backgroundColor: colors[key],
              color: getTextContrastColor(colors[key]),
            }}
            >
              <span>{key}</span>
            </div>
          ))}
        </CenteredContent>
      </Panel>
    </div>
  ));
