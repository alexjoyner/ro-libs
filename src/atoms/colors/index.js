export const colors = {
  primaryLight: '#74b9ff',
  primary: '#0984e3',
  successLight: '#55efc4',
  success: '#00b894',
  warningLight: '#ffeaa7',
  warning: '#fdcb6e',
  dangerLight: '#ff7675',
  danger: '#d63031',
};
