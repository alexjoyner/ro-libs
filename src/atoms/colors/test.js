import { colors } from './';

describe('colors', () => {
  it('Should match all default colors', () => {
    expect(colors).toMatchSnapshot();
  });
});
