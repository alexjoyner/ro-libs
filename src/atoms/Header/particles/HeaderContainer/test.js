import React from 'react';
import { shallow } from 'enzyme';
import { HeaderContainer } from './';
import { colors } from '../../../colors';

describe('HeaderContainer component', () => {
  let component;
  beforeEach(() => {
    component = shallow(<HeaderContainer />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
  it('Should display apply correct props when not sticky', () => {
    expect(component).not.toHaveStyleRule('z-index', '2');
    expect(component).not.toHaveStyleRule('position', 'fixed');
    expect(component).not.toHaveStyleRule('top', '0');
  });
  it('Should display apply correct props when sticky', () => {
    const testSticky = shallow(<HeaderContainer sticky />);
    expect(testSticky).toHaveStyleRule('z-index', '2');
    expect(testSticky).toHaveStyleRule('position', 'fixed');
    expect(testSticky).toHaveStyleRule('top', '0');
  });
  it('should accept a color prop', () => {
    const testHeight = shallow(<HeaderContainer color="primary" />);
    expect(testHeight).toHaveStyleRule('background-color', colors.primary);
  });
});
