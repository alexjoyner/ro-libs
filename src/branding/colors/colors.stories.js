import React from 'react';
import {Brand} from "../_brand.stories";
import {Panel} from "../../atoms/panel/panel";
import {COLORS} from "./colors";
import {Grid} from "../../atoms/grid/grid";

let TestColorBox = (props) => (
    <div style={{
        'height': '100px',
        'backgroundColor': props.color
    }}/>
);

Brand
    .add('Colors', () => {
        return (
            <div>
                <Panel>
                    <h2>Testing</h2>
                    <Grid columns={2}>
                        <TestColorBox color={COLORS.primary}/>
                        <TestColorBox color={COLORS.primaryLight}/>
                        <TestColorBox color={COLORS.success}/>
                        <TestColorBox color={COLORS.successLight}/>
                        <TestColorBox color={COLORS.warning}/>
                        <TestColorBox color={COLORS.warningLight}/>
                        <TestColorBox color={COLORS.danger}/>
                        <TestColorBox color={COLORS.dangerLight}/>
                    </Grid>
                </Panel>
            </div>
        )
    });