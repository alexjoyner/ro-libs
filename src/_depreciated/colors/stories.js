import React from 'react';
import { Atoms } from '../../atoms/_atoms.stories';
import { Panel } from '../../atoms/Panel';
import { colors } from '.';
import { CenteredContent } from '../CenteredContent';
import { getTextContrastColor } from '../../utils/getTextContrastColor';

Atoms
  .add('Colors', () => (
    <div>
      <Panel>
        <CenteredContent>
          <h2>Component library color pallete</h2>
          {Object.keys(colors).map(key => (
            <div style={{
              height: '100px',
              width: '100%',
              backgroundColor: colors[key],
              color: getTextContrastColor(colors[key]),
            }}
            >
              <span>{key}</span>
            </div>
          ))}
        </CenteredContent>
      </Panel>
    </div>
  ));
