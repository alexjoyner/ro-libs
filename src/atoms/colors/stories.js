// @flow
import React from 'react';
import { Atoms } from '../_atoms.stories';
import { Panel } from '../../atoms/Panel';
import { colors } from './';
import { CenteredContent } from '../../atoms/CenteredContent';
import { getTextContrastColor } from '../../utils/getTextContrastColor';

type Props = {
  color: string,
  name: string,
}
const TestColorBox = (props: Props) => (
  <div style={{
    height: '100px',
    width: '100%',
    backgroundColor: props.color,
    color: getTextContrastColor(props.color),
  }}
  >
    <span>{props.name}</span>
  </div>
);

Atoms
  .add('Colors', () => (
    <div>
      <Panel>
        <CenteredContent>
          <h2>Component library color pallete</h2>
          {Object.keys(colors).map(key => (<TestColorBox
            key={key}
            color={colors[key]}
            name={key}
          />))}
        </CenteredContent>
      </Panel>
    </div>
  ));

