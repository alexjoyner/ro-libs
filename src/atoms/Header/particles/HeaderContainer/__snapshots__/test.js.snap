// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer component Should match snapshot 1`] = `
.c0 {
  padding: 1px 5%;
  width: 100%;
  background-color: #b2bec3;
  color: black;
}

<div
  className="c0"
/>
`;
